src/coreclr/vm/CMakeLists.txt

```diff
@@ -627,6 +632,7 @@ if(CLR_CMAKE_TARGET_WIN32)
 
 if(CLR_CMAKE_TARGET_ARCH_AMD64)
     set(VM_SOURCES_WKS_ARCH_ASM
+        ${ARCH_SOURCES_DIR}/AllocSlow.asm
```

Comment:
Traditionally, each slow helper corresponds 1:1 with a fast helper, and in assembly, we inline-tail to the slow path. Is that the case here as well? If the slow and fast helpers are mutually exclusive, we could consider placing them in the same file without the slow/fast qualifier.

Reply:
These slow variants are CoreCLR specific which is why I split them off.

Reply:
WriteBarriers is using `#ifdef FEATURE_NATIVEAOT` conditions. Usually slow fallback is a call to a C++ function. Maybe `Alloc.CoreCLR.S` is a better split than slow/fast suffix.